arangod/Utils/SupportInfoBuilder.cpp

```diff
@@ -0,0 +1,762 @@
+#include "Cluster/ServerState.h"
+#include "GeneralServer/AuthenticationFeature.h"
+#include "Indexes/Index.h"
+#include "Metrics/MetricsFeature.h"
+#include "Network/Methods.h"
+#include "Network/NetworkFeature.h"
+#include "Replication/ReplicationFeature.h"
+#include "Rest/Version.h"
+#include "RestServer/CpuUsageFeature.h"
+#include "RestServer/DatabaseFeature.h"
+#include "RestServer/EnvironmentFeature.h"
+#include "RestServer/ServerIdFeature.h"
+#include "Statistics/ServerStatistics.h"
+#include "StorageEngine/EngineSelectorFeature.h"
+#include "StorageEngine/StorageEngine.h"
+#include "Transaction/StandaloneContext.h"
+#include "Utils/DatabaseGuard.h"
+#include "Utils/SingleCollectionTransaction.h"
+#include "VocBase/LogicalCollection.h"
+#include "VocBase/LogicalView.h"
+#include "VocBase/Methods/Collections.h"
+#include "VocBase/Methods/Databases.h"
+#include "VocBase/Methods/Indexes.h"
+
+#include <velocypack/Builder.h>
+#include <absl/strings/str_replace.h>
+
+#include "Logger/LogMacros.h"
+
+using namespace arangodb;
+using namespace arangodb::rest;
+using namespace std::literals;
+
+namespace {
+network::Headers buildHeaders() {
+  auto auth = AuthenticationFeature::instance();
+
+  network::Headers headers;
+  if (auth != nullptr && auth->isActive()) {
+    headers.try_emplace(StaticStrings::Authorization,
+                        "bearer " + auth->tokenCache().jwtToken());
+  }
+  return headers;
+}
+}  // namespace
+
+void SupportInfoBuilder::addDatabaseInfo(VPackBuilder& result,
+                                         VPackSlice infoSlice,
+                                         ArangodServer& server) {
+  DatabaseFeature& dbFeature = server.getFeature<DatabaseFeature>();
+
+  std::vector<std::string> databases = methods::Databases::list(server, "");
+
+  containers::FlatHashMap<std::string_view, uint32_t> dbViews;
+  for (auto const& database : databases) {
+    auto vocbase = dbFeature.lookupDatabase(database);
+
+    LogicalView::enumerate(*vocbase,
+                           [&dbViews, &database = std::as_const(database)](
+                               LogicalView::ptr const& view) -> bool {
+                             dbViews[database]++;
+
+                             return true;
+                           });
+  }
+
+  struct DbCollStats {
+    size_t numDocColls;
+    size_t numGraphColls;
+    size_t numSmartColls;
+    size_t numDisjointGraphs;
+    VPackBuilder builder;
+  };
+
+  std::unordered_map<std::string_view, DbCollStats> visitedDatabases;
+  containers::FlatHashMap<size_t, containers::FlatHashSet<std::string_view>>
+      visitedColls;
+  containers::FlatHashMap<size_t, size_t> collNumDocs;
+  // merge all collections belonging to the database into
+  // the same object, as the database might be in more
+  // than one dbserver
+  for (auto const dbItFromServers : velocypack::ObjectIterator(infoSlice)) {
+    for (auto const dbIt : velocypack::ArrayIterator(dbItFromServers.value)) {
```

Comment:
No need to make const here
```suggestion
  for (auto dbItFromServers : velocypack::ObjectIterator(infoSlice)) {
    for (auto dbIt : velocypack::ArrayIterator(dbItFromServers.value)) {
```